### PR TITLE
Separate `skdb/wasm-b64` exports entrypoint

### PR DESCRIPTION
**Pull request by schickling**
Usage:

```ts
import { SKDB, fetchWasmSource } from 'skdb'

const skdb = await SKDB.create('db-name', fetchWasmSource)
```

### CODE REVIEW

**Review by gregsexton — CHANGES_REQUESTED** [review]
Factoring out the b64 string to another file is great. But I don't like that sourcing WASM has been surfaced to the user as part of the interface. Particularly not as a base64 string, this is really an implementation detail.

I'd prefer that users can pass in a function that produces the Uint8Array of wasm bytes, for SKDB to load and orchestrate. It should default to getWasmSource if not provided.

getWasmSource can be responsible for getting hold of the wasmBase64 string and doing what it can to produce bytes. Falling back to a desperate 'fetch' if needed.

**Comment by gregsexton** [comment]
My guess is that you just don't import the wasmBase64 now and rely on the fallback behaviour?

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 .PHONY: npm
 npm: clean
-	tsc --build tsconfig.all.json
+	yarn tsc --build tsconfig.all.json
```
[the remark this answers is not in the document]
Comment (schickling) [reply Makefile:3]: Use the local version of the TS compiler and doesn't assume `tsc` to be globally installed (which can lead to nasty version incompatibility problems)

File: inlineWasm.sh [hunk 2]
```diff
@@ -13,9 +13,9 @@ else
     b64cmd="base64 -w0"
 fi
 
-(echo -n "const wasmBase64 = \`"
+(echo -n "export const wasmBase64 = \`"
  $b64cmd $WASM
  echo "\`;"
- sed "s/const wasmBase64.*;//" $SOURCE) > "$TMP"
+ sed "s/export const wasmBase64.*;//" $SOURCE) > "$TMP"
```
[the remark this answers is not in the document]
Comment (schickling) [reply inlineWasm.sh:19]: Good point. Feel free to adjust as you see fit.

**Comment by schickling** [comment]
> My guess is that you just don't import the wasmBase64 now and rely on the fallback behaviour?

Yes, the `wasmBase64` behaviour would need to be used explicitly as seen in my initial comment.

**Comment by gregsexton** [comment]
Looks good. I need to transplant this to an upstream repo otherwise it will get clobbered. Will do that now. So I'm closing this and you'll see the changes get pushed through with the next release.

**Pull request closed without merging**